### DISCUSSION TITLE
Size comparison values in table

### DISCUSSION THREAD

**Discussion by mirouml** [flatgeobuf/flatgeobuf#408] [General]
I do not understand Size comparison values. Shapefile 1 and FlatGeobuf 0.77 I would assume is fine. But GeoJSON 1.2 and GML 2.1? No way. Unless we are talking gzip or such?

**Comment by bjornharrtell** [comment]
@mirouml no compression and I just re-verified and numbers are correct as far as I can tell. Are you assuming the overhead should be larger for GeoJSON and GML? It can probably be depending on data, but in this case it's 1.2x and 2.1x. Neither shapefile or FlatGeobuf are space optimized formats (but still wins because less redundancy and binary encoding).

**Reply by mirouml** [reply]
@bjornharrtell Thank you. Yes, in my books if I convert shepefile (or FlatGeobuf) to GeoJSON the output is hugely larger due to GeoJSON being text file and with tags for each feature, attribute value etc.I would expect the difference to be similar to the read speed - about order of magnitude. It sure depends how long the text values are compared to overhead of tags, coordinates precision, geometries complexity etc. Still only 20% difference does not look right to me.

**Reply by bjornharrtell** [reply]
I think your expectations are wrong. The size difference would only be an order of magnitude if they redundancy is an order of magnitude. The actual data is not much larger.

**Reply by mirouml** [reply]
You are right, my expectations are skewed by the form of data I usually work with - many attributes with number values. I did run tests on two different freely available datasets and the results show the sizes in table in this repo should be correct:

Australia roads downloaded as gis_osm_roads_free_1 from geofabrik.de
Shapefile 1.17 GB
GeoPackage 0.99 GB
FlatGeobuf 0.99 GB
GeoJSON 1.71 GB
GML 3.58 GB

Cadastral data weekly - whole of State Queensland downloaded from qldspatial
Shapefile 3.71 GB
GeoPackage 2.00 GB
FlatGeobuf 2.32 GB
GeoJSON 4.38 GB
GML 7.58 GB

Thank you for your time, I did learn that GeoJSON is not as bad as I thought in size comparison for many (and likely typical) datasets. Hope the stats I provided here might help somebody else to learn something too. FlatGeobuf is great, love the idea of it, amazing work!
